File: buyAndSellStock_ii.py
#O(n) time and O(1) solution
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        left = 0
        right = left + 1
        n = len(prices) - 1
        #this is to avoid repeated profit calculation
        i = 0
        while i < n:
           
            while(i < n and prices[i+1] <= prices[i]):
                i += 1
            left = prices[i]
            while(i < n and prices[i + 1] > prices[i]):
                i += 1
            right = prices[i]
            profit += right - left
        
        return profit

File: BuyandSellStockII.java
// Time Complexity : O(n) 
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this: None

// Your code here along with comments explaining your approach

class BuyAndSellStockII {
    int sum =0;
    public int maxProfit(int[] prices) {
        for(int i=1;i<prices.length;i++){
            int profit = prices[i]-prices[i-1];
            if(profit>0){
                sum+=profit;
            }
        }
        return sum;
    }
}